packages/sn-pickers-react/src/ListPicker/ListPicker.tsx

```diff
@@ -18,21 +19,40 @@ export function ListPickerComponent<T extends GenericContent = GenericContent>(p
   const [currentPath, setCurrentPath] = useState(currentContentPath)
   const [parentId, setParentId] = useState<number | undefined>(props.parentId)
   const [selectedId, setSelectedId] = useState<string | number>(0)
+  const { data: items, error: itemsError, isLoading: areItemsLoading, reload } = useAsync({
+    promiseFn: loadItems,
+    path: currentPath,
+    repository: props.repository,
+    oDataOptions: props.itemsOdataOptions,
+    watch: currentPath,
+  })
+  const { data: parent, error: parentError, isLoading: isParentLoading } = useAsync({
+    promiseFn: loadParent,
+    repository: props.repository,
+    id: parentId,
+    oDataOptions: props.parentODataOptions,
+    watch: parentId,
+  })
 
-  const loadItems = () =>
-    (props.loadItems && props.loadItems(currentPath)) || defaultLoadItems(currentPath, props.repository)
-
-  const loadParent = () =>
-    (props.loadParent && props.loadParent(parentId)) || defaultLoadParent(parentId, props.repository)
-
-  const { loading, value: items, error } = useAsync(loadItems, [currentPath])
-  const parent = useAsync(loadParent, [parentId])
+  const update = () => reload()
```

Comment:
There can be many events that triggers unneccessary reloads (e.g. uploading many small files). Can we debounce reload to improve performance?

Reply:
I've added debounce there.